test/ffi/datasets.jl

```diff
@@ -144,9 +144,40 @@ end
 
 
 @testset "LGBM_DatasetSetField" begin
-    # The current implementation for LGBM_DatasetSetField has questionable type promotions
-    # but it touches too much code in order to tidy up right now
-    @test_broken false
+    mymat = [1. 2.; 3. 4.; 5. 6.]
+    ds = LightGBM.LGBM_DatasetCreateFromMat(mymat, verbosity)
+
+    @testset "_LGBM_DatasetSetField invalid field name test" begin
```

Comment:
This test is essentially checking the type conversions and field allocation which is what the function is doing.